src/components/Timer.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState, Fragment } from "react";
+import styled from "styled-components";
+import {
+  TypographyProps,
+  typography,
+  FlexboxProps,
+  flexbox,
+  SpaceProps,
+  space,
+} from "styled-system";
+
+const Container = styled.div<FlexboxProps & SpaceProps & TypographyProps>`
+  ${flexbox};
+  ${space};
+  ${typography};
+  display: flex;
+`;
+
+const H1 = styled.h1<TypographyProps & FlexboxProps>`
+  ${typography};
+  ${flexbox};
+  text-transform: uppercase;
+`;
+
+const Timer = () => {
+  const calculateTimeLeft = () => {
+    let year = new Date().getFullYear();
+    const difference = +new Date(`${year}-09-1`) - +new Date();
+    let timeLeft = {};
+
+    if (difference > 0) {
+      timeLeft = {
+        days: Math.floor(difference / (1000 * 60 * 60 * 24)),
+        hours: Math.floor((difference / (1000 * 60 * 60)) % 24),
+        minutes: Math.floor((difference / 1000 / 60) % 60),
+        seconds: Math.floor((difference / 1000) % 60),
+      };
+    }
+    return timeLeft;
+  };
+
+  const [timeLeft, setTimeLeft] = useState<any>(calculateTimeLeft());
+  const [year] = useState(new Date().getFullYear());
+
+  useEffect(() => {
+    setTimeout(() => {
+      setTimeLeft(calculateTimeLeft());
+    }, 1000);
+  });
+
+  const timerComponents: {} | null | any = [];
```

Comment:
Using `any` is a bit of a last resort ...
Perhaps at the top of the file we could define some types:
```
interface TimeLeft {
  days: number;
  hours: number;
  minutes: number;
  seconds: number;
}
```
then
```
const [timeLeft, setTimeLeft] = useState<TimeLeft>(calculateTimeLeft());
```
then could have a helper function:
```
const hasReachedEnd = (timeLeft: TimeLeft) => {
  // some logic

  return true;
  // return false; 
} 
```

This way we keep the TimeLeft more typesafe

Reply:
turns out the `hasReachedEnd ` function is superfluous since the page will be replace before the timer reaches 0, and if not then the timer will show "0 days 0 hours ..." which is fine too

Reply:
the other changes have been made though 👍